module/zfs/dsl_scan.c

```diff
@@ -2038,6 +2038,26 @@ dsl_scan_visitbp(blkptr_t *bp, const zbookmark_phys_t *zb,
 		return;
 	}
 
+	/*
+	 * Check if this block contradicts any filesystem flags.
+	 */
+	spa_feature_t f = SPA_FEATURE_LARGE_BLOCKS;
+	if (BP_GET_LSIZE(bp) > SPA_OLD_MAXBLOCKSIZE)
+		ASSERT3B(dsl_dataset_feature_is_active(ds, f), ==, B_TRUE);
```

Comment:
nit: Might as well use `ASSERT()` (likewise below)

Reply:
without `== B_TRUE` would be even better

Reply:
fixed!